lib/Sema/CSGen.cpp

```diff
@@ -3088,6 +3088,10 @@ namespace {
       return openPackElement(packType, /*packRepr*/ nullptr);
     }
 
+    Type visitMaterializePackExpr(MaterializePackExpr *expr) {
+      llvm_unreachable("MaterializePackExpr already type-checked");
```

Comment:
Old code completion implementation can still (partially) re-typecheck previously type-checked AST. Could you please update `SanitizeExpr` to transform `MaterializePackExpr` back into the original form?

Reply:
I suspect code completion paths that go through `SanitizeExpr` and involve variadic generics expression nodes are busted due to `PackExpansionExpr` and `PackElementExpr` too. I'll put together some test cases and fix the issues holistically in a follow-up PR.

Reply:
Sounds good!